modules/azure/nsg_test.go

```diff
@@ -0,0 +1,146 @@
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package azure
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+)
+
+func TestPortRangeParsing(t *testing.T) {
+	var cases = []struct {
+		portRange    string
+		expectedLo   int
+		expectedHi   int
+		expectsError bool
+	}{
+		{"22", 22, 22, false},
+		{"22-80", 22, 80, false},
+		{"*", 0, 65535, false},
+		{"*-*", 0, 0, true},
+		{"22-", 0, 0, true},
+		{"-80", 0, 0, true},
+		{"-", 0, 0, true},
+		{"80-22", 22, 80, false},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.portRange, func(t *testing.T) {
+			lo, hi, err := parsePortRangeString(tt.portRange)
+			if !tt.expectsError {
+				require.NoError(t, err)
+			}
+			assert.Equal(t, tt.expectedLo, int(lo))
+			assert.Equal(t, tt.expectedHi, int(hi))
+		})
+	}
+}
+
+func TestNsgRuleSummaryConverstion(t *testing.T) {
+	// Quick test to make sure the safe nil handling is working
+	var name = "test name"
+	var sdkStruct = network.SecurityRulePropertiesFormat{}
+
+	// Verify the nil values were correctly defaulted to "" without a panic
+	result := convertToNsgRuleSummary(&name, &sdkStruct)
+	assert.Equal(t, "", result.Description)
+	assert.Equal(t, "", result.SourcePortRange)
+	assert.Equal(t, "", result.DestinationPortRange)
+	assert.Equal(t, "", result.SourceAddressPrefix)
+	assert.Equal(t, "", result.DestinationAddressPrefix)
+	assert.Equal(t, int32(0), result.Priority)
+}
+
+func TestAllowSourcePort(t *testing.T) {
+	var cases = []struct {
+		CaseName        string
+		SourcePortRange string
+		Access          string
+		TestPort        string
+		Result          bool
+	}{
+		{"22 allowed", "22", "Allow", "22", true},
+		{"22 denied", "22", "Deny", "22", false},
+		{"22 doesn't allow 80", "22", "Allow", "80", false},
+		{"Any allows any", "*", "Allow", "*", true},
```

Comment:
Add test cases for ranges.

Reply:
Added range cases for both "Allow" and "Deny".